packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1441,10 +1822,45 @@ export class TransactionController extends BaseControllerV1<
    * Ensure `releaseLock` is called once processing of the `nonce` value is complete.
    *
    * @param address - The hex string address for the transaction.
+   * @param networkClientId - The network client ID for the transaction, used to fetch the correct nonce tracker.
    * @returns object with the `nextNonce` `nonceDetails`, and the releaseLock.
    */
-  async getNonceLock(address: string): Promise<NonceLock> {
-    return this.nonceTracker.getNonceLock(address);
+  async getNonceLock(
```

Comment:
This entire method feels like a great candidate for the potential `MultichainHelper` also...